core/src/main/java/tech/pegasys/eth2signer/core/multikey/metadata/SignerParser.java

```diff
@@ -10,11 +10,13 @@
  * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
  * specific language governing permissions and limitations under the License.
  */
-package tech.pegasys.eth2signer.core.multikey;
+package tech.pegasys.eth2signer.core.multikey.metadata;
 
-import tech.pegasys.eth2signer.core.multikey.metadata.UnencryptedKeyMetadataFile;
 import tech.pegasys.eth2signer.core.signing.ArtifactSigner;
 
-public interface MultiSignerFactory {
-  ArtifactSigner createSigner(UnencryptedKeyMetadataFile metadataFile);
+import java.nio.file.Path;
+
+public interface SignerParser {
+
+  ArtifactSigner parse(final Path file) throws SigningMetadataException;
```

Comment:
no need for "final" on an interface (apparently)

Reply:
done